swift/Sources/FlatBuffers/Message.swift

```diff
@@ -0,0 +1,27 @@
+public protocol FlatBufferGRPCMessage {
+    var rawPointer: UnsafeMutableRawPointer { get }
+    var size: Int { get }
+    
+    init(byteBuffer: ByteBuffer)
+}
+
+public final class Message<T: FlatBufferObject>: FlatBufferGRPCMessage {
+    internal var buffer: ByteBuffer
+    
+    public var object: T {
+        T.init(buffer, o: Int32(buffer.read(def: UOffset.self, position: buffer.reader)) + Int32(buffer.reader))
+    }
+    
+    public var rawPointer: UnsafeMutableRawPointer { return buffer.memory.advanced(by: buffer.reader) }
+    
+    public var size: Int { return Int(buffer.size) }
+    
+    public init(byteBuffer: ByteBuffer) {
+        buffer = byteBuffer
+    }
+    
+    public init(builder: inout FlatBufferBuilder) {
+        buffer = builder.sizedBuffer
+        builder.clear()
+    }
+}
```

Comment:
@aardappel this will be the message that will be sent from the Flatbuffers implementation to GRPC, the `FlatBufferGRPCMessage` will conform to a protocol `GRPCPayload` which will allow us to copy the ByteBuffer from the FlatBuffers implementation to the NIO.ByteBuffer

Reply:
Ok, so that holds both the buffer and a root pointer? I guess that is ok.

Reply:
@aardappel Yep, it would, and I think it would be nice to have an assert that makes sure that the developers won’t use builder.finish() before passing it to the message. What do you think?

Reply:
Not sure I am following.. why must they not call `finish` ?

Reply:
It’s just that they will just finish the buffer without buffer length prefix nor the file name prefix that can be added in flatbuffers. Since the message.object current code assumes that the first 4 bytes are the root of the object being sent. Like this we limit the developer to actually create the buffer and we finish the buffer in the message object. It’s just a suggestion after all

Reply:
I'd suggest if this needs a size prefix that that is handled separetely, since it is quite unnatural for an API to say "give me an unfinished buffer". I don't think gRPC should even require the message themselves to have a size in them anyway?

Reply:
Okay sounds good then. I'll keep it as it is.